packages/api/src/modules/bot/CreateBot.ts

```diff
@@ -0,0 +1,62 @@
+import User from '@entities/User';
+import { ContextType } from '@root/apolloServer';
+import { NoBot } from '@utils/auth/NoBot';
+import { ApolloError } from 'apollo-server-express';
+import {
+  Arg,
+  Authorized,
+  Ctx,
+  Field,
+  InputType,
+  Mutation,
+  Resolver,
+} from 'type-graphql';
+
+@InputType()
+class CreateBotInput {
+  @Field()
+  username: string;
+
+  @Field()
+  name: string;
+}
+
+@Resolver()
+export class CreateBotResolver {
+  @Mutation(() => Boolean)
+  @Authorized()
+  @NoBot()
+  async createBot(
+    @Arg('data') data: CreateBotInput,
+    @Ctx() { getUser }: ContextType
+  ) {
+    const user = await getUser();
+    const bot = User.create();
+
+    bot.isBot = true;
+    bot.avatar = user.avatar;
+
+    const sameNameUser = await User.createQueryBuilder()
+      .where('LOWER(username) = LOWER(:username)', { username: data.username })
+      .getCount();
+    if (sameNameUser !== 0) {
+      return new ApolloError(
+        'Username is taken please request a safe Username',
+        'USERNAME_TAKEN'
+      );
+    }
+
+    bot.username = data.username;
+    bot.name = data.name || data.username;
+    bot.createdAt = String(Date.now());
+    bot.verified = user.verified;
```

Comment:
Do we want to auto verify bots if the owner is verified?

Reply:
Bots should meet a certain requirement to get verified, so no.